tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs

```diff
@@ -189,8 +189,7 @@ internal static string CreateTraceStateHeader(SpanContext context)
                     sb.Length--;
                 }
 
-                // additional tracestate from other vendors
-                var additionalState = context.TraceContext?.AdditionalW3CTraceState;
+                var additionalState = context.AdditionalW3CTraceState;
```

Comment:
For other reviewers, this is the same deal as origin, but required a `SpanContext` change in this PR to make sure that `SpanContext.AdditionalW3CTraceState` checks `TraceContext.AdditionalW3CTraceState` as part of the getter logic